pytorchfi/core.py

```diff
@@ -167,26 +177,28 @@ def declare_weight_fi(self, **kwargs):
         corrupt_idx = [corrupt_k, corrupt_c, corrupt_kH, corrupt_kW]
 
         current_layer = 0
-        for name, param in self.corrupted_model.named_parameters():
-            if "weight" in name and ("features" in name or "conv" in name):
-                if current_layer == corrupt_layer:
-                    corrupt_idx = (
-                        tuple(corrupt_idx)
-                        if isinstance(corrupt_idx, list)
-                        else corrupt_idx
-                    )
-                    orig_value = param.data[corrupt_idx].item()
-                    if custom_injection:
-                        corrupt_value = CUSTOM_FUNCTION(param.data, corrupt_idx)
-                    param.data[corrupt_idx] = corrupt_value
-
-                    logging.info("Weight Injection")
-                    logging.info("Layer index: %s", corrupt_layer)
-                    logging.info("Module: %s", name)
-                    logging.info("Original value: %s", orig_value)
-                    logging.info("Injected value: %s", corrupt_value)
-
-                current_layer += 1
+        for layer in self.corrupted_model.modules():
+            for i in self.get_inj_layer_types():
+                if isinstance(layer, i):
```

Comment:
Can condense these two conditions to one line

Reply:
This can be cleaned up a bit I agree, but we can't directly combine the two conditions, specifically because of the `current_layer += 1` statement.

Reply:
turns out there was a bigger issue here. Fixed with recent commit.